kimchi/src/circuits/constraints.rs

```diff
@@ -961,4 +962,54 @@ pub mod tests {
             assert_eq!(res.domain.d1.size, expected_domain_size)
         }
     }
+
+    #[test]
+    fn test_lookup_domain_size_computation() {
+        let (next_start, range_check_gates_0) = CircuitGate::<Fp>::create_range_check(0); /* 1 range_check gate */
+        let (next_start, range_check_gates_1) = CircuitGate::<Fp>::create_range_check(next_start); /* 1 range_check gate */
+        let (next_start, xor_gates_0) = CircuitGate::<Fp>::create_xor_gadget(next_start, 3); /* 1 xor gate */
+        let (next_start, xor_gates_1) = CircuitGate::<Fp>::create_xor_gadget(next_start, 3); /* 1 xor gate */
+        let (_, ffm_gates) =
+            CircuitGate::<Fp>::create_foreign_field_mul(next_start, &Fq::modulus_biguint()); /* 1 foreign field multiplication gate */
+        let circuit_gates: Vec<CircuitGate<Fp>> = range_check_gates_0
+            .into_iter()
+            .chain(range_check_gates_1)
+            .chain(xor_gates_0)
+            .chain(xor_gates_1)
+            .chain(ffm_gates)
+            .collect(); /* 2 range check gates + 2 xor gates + 1 foreign field multiplication */
+
+        // inputs + expected output
+        let data = [
+            (
+                (10, 10),
+                8192, /* 8192 > 10 * 10 + 1 * 4096 + 1 * 256 + 1 + zk_row */
+            ),
+            (
+                (0, 0),
+                8192, /* 8192 > 0 * 0 + 1 * 4096 + 1 * 256 + 1 + zk_row */
+            ),
+            (
+                (5, 100),
+                8192, /* 8192 > 5 * 100 + 1 * 4096 + 1 * 256 + 1 + zk_row */
+            ),
+        ];
+        for ((number_of_table_ids, size), expected_domain_size) in data.into_iter() {
```

Comment:
Nit: personal preference to use `data.into_iter().for_each()` for a functional approach.

Reply:
Done https://github.com/o1-labs/proof-systems/pull/2369/commits/32637674c1975382ae82179db22b9324a11f8c43